### DISCUSSION TITLE
document.getElementById("root") in next.js

### DISCUSSION THREAD

**Discussion by enkicoma** [vercel/next.js#17978] [Help] [answered]
Hi I am facing an issue with scss and next.js

in a simple reacts app this does work

```
ReactDOM.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
  document.getElementById("root")
);
```

scss:

```
html,
body,
#root {
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  overflow: hidden;
}

```
In next.js, it doesn't work... next.js doesn't have a `#root` id element and all my scss doesn't work in UI it renders everything in one main div block, everything overlaps...

_app.js:

```
import './index.scss'

function MyApp({ Component, pageProps }) {
  return <Component {...pageProps} />
}

export default MyApp
```

How can I resolve this issue? I need the same thing `document.getElementById("root")` because in react native works..

**Comment by enkicoma** [comment, marked as answer]
found it... instead of refrecing to `#root`  it requires to be `#__next`
```
html,
body,
#__next {
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  overflow: hidden;
}
```

**Reply by RachaelSMathew** [reply]
Hi, I'm a little confused. How would you do `document.getElementById("idValue").checked`? 
I keep getting the error that `document cannot be defined`.

**Reply by enkicoma** [reply]
I am not sure what you doing but this may help:
https://github.com/vercel/next.js/discussions/24305

**Reply by RachaelSMathew** [reply]
I am using Next JS, and I am trying to `document.getElementById` and `document.getElementsByClassName`, but neither of these is working because `document is undefined`.
I have gone to #24305 and 
```
useEffect(() => {

  if (document) {
    document.getElementById("#__next").style.height = "100vh";
  }

}, []);
```

produced the same error that `document is undefined `
<img width="978" alt="Screen Shot 2022-01-14 at 2 36 31 PM" src="https://user-images.githubusercontent.com/30049533/149574861-5ebc8c10-1d1d-4b76-bf76-8fb95925bc7e.png">

**Reply by mattcarlotta** [reply]
@RachaelSMathew When using [getElementById](https://developer.mozilla.org/en-US/docs/Web/API/Document/getElementById) you don't include the hash `#` (`#example` just becomes `example`). Similarly, when using [getElementsByClassName](https://developer.mozilla.org/en-US/docs/Web/API/Document/getElementsByClassName) you don't include the dot `.` (`.example` just becomes `example`) to select the classes. On a related note, the `getElementsByClassName` method returns an HTMLCollection so you would need to specify the [index](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection#methods) of the node you want to select. 

Working codesandbox [here](https://codesandbox.io/s/nextjs-document-id-selection-gglsw)

```jsx
import { useEffect } from "react";

export default function IndexPage() {
  useEffect(() => {
    if (document) {
      /*
         To avoid potential crashes of null/undefined selections, you may want 
         to save the selection as a variable and check if the selection exists  
         before attempting to add a style. For example, type guarding the 
         selection to make sure its defined before attempting to access 
         Element properties:
 
         const node = document.getElementById("container");
         if (node) node.style = "...";
      */ 
      document.getElementById("container").style =
        "height: calc(100vh - 120px);background: black;color:white;display:flex;justify-content:center;align-items:center;";
    }
  }, []);

  return (
    <div>
      <h1 style={{ textAlign: "center" }}>ID Select Example</h1>
      <div id="container">
        <h1>Container</h1>
      </div>
    </div>
  );
}

```

If you don't need to dynamically style an element, then I'd recommend using a [global stylesheet](https://nextjs.org/docs/basic-features/built-in-css-support#adding-a-global-stylesheet) instead.

**Reply by RachaelSMathew** [reply]
What if I'm calling document in a JavaScript function right before `return()`?
Like this:
```
    function example() {
        document.getElementsByClassName("changeTextExperiences")[0].classList.toggle('hideExperiences');
    }
```

<img width="980" alt="Screen Shot 2022-01-14 at 5 20 29 PM" src="https://user-images.githubusercontent.com/30049533/149592716-85509def-0179-4523-ad2c-ef93aefca86e.png">

**Reply by mattcarlotta** [reply]
@RachaelSMathew

The problem is described in the error: You're trying to access the property `classList` on something that's `undefined` -- your selection `document.getElementsByClassName("changeTextExperiences")` appears to be returning an empty collection. See my comments above about type guarding (comments in the `useEffect`).

On a related note, you can use React state to conditionally add/remove a className (in this case, toggling between `show-experiences` and `hide-experiences` classes):

```jsx
const Example = () => {
  const [showExperiences, setExperiencesVisibility] = React.useState(false);

  return (
    <>
      <div className={`${showExperiences ? "show" : "hide"}-experiences`}>
        Example Experiences
      </div>
      <button type="button" onClick={() => setExperiencesVisibility(prevState => !prevState)}>
        Toggle Experiences
      </button>
    </>
  );
}
```

**Comment by Ikimonogatari** [comment]
const selectLensElements = Array.from(
      document.getElementsByClassName("selectLens")
    );
It doesn't work when deployed. But works fine on local. What is the problem ?
